### PR TITLE
Rpb title toggle

### LINKED ISSUE #695: Report builder to toggle between using applied and legal titles
Much similar to IGOC and estimates comparison tool

### PR DESCRIPTION
**Pull request by AmndaPlant**
Closes #695 

Allows for the report builder to toggle between legal and applied titles. I was unsure of where to put the button (it's currently in the "Select your data" box underneath the "Select another table or topic" button) but I'm open to suggestions.

### CODE REVIEW

**Comment by ktw1016** [comment]
See my comment in Slack!

**Comment by AmndaPlant** [comment]
Recent commits should close #720

File: client/src/components/DisplayTable/DisplayTable.js [hunk 1]
```diff
@@ -93,6 +94,7 @@ export class DisplayTable extends React.Component {
           sum_func: (sum, value) => ... <- (function) Custom sum func. Default to sum + value
           sort_func: (a, b) => ... <- (function) Custom sort func. Default to _.sortBy
           sum_initial_value: 0 <- (number) Default to 0
+          is_toggleable: true <- (boolean) Default to true
```
Comment (ktw1016) [inline client/src/components/DisplayTable/DisplayTable.js:97]: I think more explicit name should be used like `visibility_toggleable` or something like that.
Also, please checkout `column_config_defaults`. You shouldn't need to set this flag to `false` everywhere. Setting it `false` in defaults variable will do it.

I'll review the rest of the code after you make this change!

File: client/src/components/DisplayTable/DisplayTable.js [hunk 2]
```diff
@@ -93,6 +99,7 @@ export class DisplayTable extends React.Component {
           sum_func: (sum, value) => ... <- (function) Custom sum func. Default to sum + value
           sort_func: (a, b) => ... <- (function) Custom sort func. Default to _.sortBy
           sum_initial_value: 0 <- (number) Default to 0
+          visibility_toggleable: true <- (boolean) Default to true
```
Comment (ktw1016) [inline client/src/components/DisplayTable/DisplayTable.js:102]: Please update this documentation

File: client/src/components/DisplayTable/DisplayTable.js [hunk 3]
```diff
@@ -44,6 +44,12 @@ export class DisplayTable extends React.Component {
       (supplied_column_config) => ({
         ...column_config_defaults,
         ...supplied_column_config,
+        // Set visibility_toggleable default based off index
```
Comment (ktw1016) [inline client/src/components/DisplayTable/DisplayTable.js:47]: No need to check if it's defined. Also no need for ternary here

```
    const col_configs_with_defaults = _.mapValues(
      column_configs,	      column_configs,
      (supplied_column_config) => ({	      (supplied_column_config) => ({
        // Set visibility_toggleable default based off index
        visibility_toggleable: supplied_column_config.index !== 0
        ...column_config_defaults,	        ...column_config_defaults,
        ...supplied_column_config,	        ...supplied_column_config,
      })
    );
```
So if `visibility_toggleable` was manually passed in, object spread operator ` ...supplied_column_config` will override the value.

File: client/src/rpb/index.js [hunk 4]
```diff
@@ -96,13 +96,15 @@ class RPB extends React.Component {
     if (props.state.table) {
       this.state = {
         table_picking: false,
+        use_legal_titles: false,
```
Comment (ktw1016) [inline client/src/rpb/index.js:99]: all of this seem dead code?

**Comment by AmndaPlant** [comment]
Whoops my mistake, forgot to update the documentation and forgot there's two places where `col_configs_with_defaults` is defined

**Pull request merged**